systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,499 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+            .withNewTls()
+                .withTrustedCertificates(certSecretTarget)
+            .endTls()
+            .addToConfig("config.storage.replication.factor", -1)
+            .addToConfig("offset.storage.replication.factor", -1)
+            .addToConfig("status.storage.replication.factor", -1)
+            .build();
+
+        resourceManager.createResource(extensionContext, KafkaMirrorMaker2Templates.kafkaMirrorMaker2(clusterName, kafkaClusterTargetName, kafkaClusterSourceName, 1, true)
+            .editSpec()
+                .withClusters(sourceClusterWithTlsAuth, targetClusterWithTlsAuth)
+                .editFirstMirror()
+                    .withTopicsPattern(MIRRORMAKER2_TOPIC_NAME + ".*")
+                .endMirror()
+            .endSpec()
+            .build());
+        Map<String, String> mmSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+
+        LOGGER.info("Setting topic to {}, cluster to {} and changing user to {} for next internalKafkaClient.",
+            topicSourceNameA, kafkaClusterSourceName, userSource.getMetadata().getName());
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withTopicName(topicSourceNameA)
+            .withClusterName(kafkaClusterSourceName)
+            .withKafkaUsername(userSource.getMetadata().getName())
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .build();
+
+        LOGGER.info("Sending messages to - topic {}, cluster {} and message count of {}",
+            topicSourceNameA, kafkaClusterSourceName, messagesCount);
+        int sent = internalKafkaClient.sendMessagesTls();
+
+        LOGGER.info("Receiving messages from - topic {}, cluster {} and message count of {}",
+            topicSourceNameA, kafkaClusterSourceName, messagesCount);
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+
+        LOGGER.info("Now setting topic to {}, cluster to {} and user to {} - the messages should be mirrored",
+            topicTargetNameA, kafkaClusterTargetName, userTarget.getMetadata().getName());
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withTopicName(topicTargetNameA)
+            .withClusterName(kafkaClusterTargetName)
+            .withKafkaUsername(userTarget.getMetadata().getName())
+            .build();
+
+        LOGGER.info("Consumer in target cluster and topic should receive {} messages", messagesCount);
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+        LOGGER.info("Messages successfully mirrored");
+
+        LOGGER.info("[TLS] Going to delete ClientsCA certificates");
+        SecretUtils.deleteSecretWithWait(KafkaResources.clientsCaCertificateSecretName(kafkaClusterSourceName), namespaceName);
+        KafkaUtils.waitForKafkaReady(kafkaClusterSourceName);
```

Comment:
Shouldn't you wait for rolling updates rather than waiting for Kafka cluster to be ready? Once Kafka will be ready, you don't need to wait for deployment rolling update, which is on the next line.

Reply:
Fixed accordingly I believe.